flink/v1.19/flink/src/main/java/org/apache/iceberg/flink/FlinkConfigOptions.java

```diff
@@ -69,7 +69,7 @@ private FlinkConfigOptions() {}
   public static final ConfigOption<Boolean> TABLE_EXEC_ICEBERG_EXPOSE_SPLIT_LOCALITY_INFO =
       ConfigOptions.key("table.exec.iceberg.expose-split-locality-info")
           .booleanType()
-          .noDefaultValue()
+          .defaultValue(false)
```

Comment:
This is a backwards incompatible change. I'm not sure how widely it's used, but we need to bdo some research and be more vocal about this change, if we decide to go ahead with it.

Reply:
agree with @pvary that this changes the default behavior, which calls `Util.mayHaveBlockLocations(table.io(), table.location())` from Hadoop module to figure out if locality is enabled for `hdfs` scheme.

would it work if we add the `isHadoopCommonOnClasspath` check at the beginning of the `Util#mayHaveBlockLocations` class in Hadoop module? return false if Hadoop common not on class path?
```
  public static boolean mayHaveBlockLocations(FileIO io, String location) {
    if (usesHadoopFileIO(io, location)) {
      InputFile inputFile = io.newInputFile(location);
      if (inputFile instanceof HadoopInputFile) {
        String scheme = ((HadoopInputFile) inputFile).getFileSystem().getScheme();
        return LOCALITY_WHITELIST_FS.contains(scheme);

      } else {
        return false;
      }
    }

    return false;
  }
```